app/pages/lab/social-links-editor.jsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import DragReorderable from 'drag-reorderable';
+import AutoSave from '../../components/auto-save.coffee';
+import SOCIAL_ICONS from '../../lib/social-icons';
+
+export default class SocialLinksEditor extends React.Component {
+  constructor(props) {
+    super(props);
+    const socialOrder = Object.keys(SOCIAL_ICONS).map(key => key);
+    this.state = {
+      socialOrder
+    };
+    this.reorderDefault = this.reorderDefault.bind(this);
+    this.handleNewLink = this.handleNewLink.bind(this);
+    this.handleLinkReorder = this.handleLinkReorder.bind(this);
+    this.handleRemoveLink = this.handleRemoveLink.bind(this);
+    this.renderRow = this.renderRow.bind(this);
+  }
+
+  componentDidMount() {
+    this.reorderDefault();
+  }
+
+  reorderDefault() {
+    const socialUrls = this.props.project.urls.filter(url => url.path);
+    const newOrder = [];
+    socialUrls.map(link => newOrder.push(link.site));
+    this.state.socialOrder.map((item) => {
+      if (newOrder.indexOf(item) < 0) {
+        newOrder.push(item);
+      }
+    });
+    this.setState({ socialOrder: newOrder });
+  }
+
+  handleNewLink(site, e) {
+    let index = this.indexFinder(this.props.project.urls, site);
+    if (index < 0) { index = this.props.project.urls.length; }
+
+    if (e.target.value) {
+      const changes = {
+        [`urls.${index}`]: {
+          label: '',
+          path: e.target.value,
+          site,
+          url: `https://${site}${e.target.value}`
+        }
+      };
+      this.props.project.update(changes);
+    } else {
+      this.handleRemoveLink(site);
+    }
+  }
+
+  handleLinkReorder(newLinkOrder) {
```

Comment:
I liked how you used `.sort` in a different file, that I coincidentally reviewed before this one. I wasn't familiar with it, had the related docs up and it felt like could apply here. After some help from Google and a little messing around I'm not sure if the following will cause any issue (doesn't appear to, could use confirmation), but the following might work(?):
```
handleLinkReorder(newLinkOrder) {
    const externalUrls = this.props.project.urls.filter(url => !url.path);
    const socialUrls = this.props.project.urls.filter(url => url.path);
    const newSocialUrls = socialUrls.sort((a, b) => newLinkOrder.indexOf(a.site) - newLinkOrder.indexOf(b.site));
    const changes = { urls: externalUrls.concat(newSocialUrls) };
    this.props.project.update(changes).save();
    this.setState({ socialOrder: newLinkOrder });
  }
```
If think helps please feel free to rename variables.